### ISSUE TITLE
[RFE] Access to cache credentials is lost on daemon restart

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1531]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1527
- Created at 2012-09-17 19:23:27 by [simo](https://pagure.io/user/simo)
- Closed at 2020-03-24 14:20:56 as wontfix
- Assigned to nobody
---

We store cached credentials in the kernel keyring, but we create a new creds keyring session when we restart sssd.

This means on restart we will loose access to previous cached credentials.
We should use a named session cred and solve 2 issues:
how to avoid garbage collection during the time sssd is stopped.
And how to avoid root processes to easily access the ring (SeLinux on the databse could avoid access to any other root process and have a name generated randomly and saved to the databse).

### Comments

---

**Comment from [simo](https://pagure.io/user/simo) at [2012-09-18 01:02:38](https://pagure.io/SSSD/sssd/issue/1527#230286)**

What is wrong with this? The data that goes into keyring is the cleartext password. We need user cleartext creds from keyring only if we start offline and then get online. After that the creds are removed from the keyring (are they or we keep them forever?). If the system never gets online and reboots the new clear text password will be captured and stored.

So is this just the garbage collection?
I thought that only sssd can access the ring, is this not the case?
If it is not this seems to be different issue and IMO should be treated as a separate ticket related to protecting access to the keyring data. I has nothing to do with how many sessions we actually start.

---

**Comment from [simo](https://pagure.io/user/simo) at [2012-09-18 02:51:07](https://pagure.io/SSSD/sssd/issue/1527#230287)**

Replying to [comment:1 dpal]:
> What is wrong with this? The data that goes into keyring is the cleartext password. We need user cleartext creds from keyring only if we start offline and then get online. After that the creds are removed from the keyring (are they or we keep them forever?). If the system never gets online and reboots the new clear text password will be captured and stored.

This is a different use case, it is for when cifs.ko is using NTLM authentication instead of Kerberos authentication to transparently access windows shares.
In this case cifs.ko needs to be able to access creds at any time when online.

> So is this just the garbage collection?

I don't understand the context of this question.

> I thought that only sssd can access the ring, is this not the case?

cifs.ko is in kernel it can access the keyring if it wants/needs to.

> If it is not this seems to be different issue and IMO should be treated as a separate ticket related to protecting access to the keyring data. I has nothing to do with how many sessions we actually start.

See above

---

**Comment from [simo](https://pagure.io/user/simo) at [2012-09-18 15:17:43](https://pagure.io/SSSD/sssd/issue/1527#230288)**

1. First of all NTLM seems to be a corner case nowadays with 2003 AD already being 9 years old. I am not sure we should address it.
 2. It is not clear from the decryption above what exactly we are solving.
    - It seems that we need to keep user password all the time even when user is not logged in. Why? 
    - There are two problems mentioned in the original decryption: garbage collection and access to the password by root. It is unclear what the problems actually are. What garbage collection? Kernel discarding SSSD keyring during the SSSD restart? How does this work? And regarding root access to passwords, is it a problem that we can/need to address at all? The whole "password in clear for NTLM" use case seems very insecure and outdated. May be we should not do it at all.

---

**Comment from [simo](https://pagure.io/user/simo) at [2012-09-20 15:43:30](https://pagure.io/SSSD/sssd/issue/1527#230289)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.10 beta
rhbz:  => todo
summary: Access to cache credentials is lost on daemon restart => [RFE] Access to cache credentials is lost on daemon restart
type: defect => enhancement

---

**Comment from [simo](https://pagure.io/user/simo) at [2012-12-20 23:33:42](https://pagure.io/SSSD/sssd/issue/1527#230290)**

Fields changed

selected:  => Not need

---

**Comment from [simo](https://pagure.io/user/simo) at [2013-01-02 15:32:08](https://pagure.io/SSSD/sssd/issue/1527#230291)**

Moving tickets that are not a priority for SSSD 1.10 into the next release.

milestone: SSSD 1.10 beta => SSSD 1.11 beta

---

**Comment from [simo](https://pagure.io/user/simo) at [2014-09-26 18:49:56](https://pagure.io/SSSD/sssd/issue/1527#230292)**

Fields changed

mark:  => 0

---

**Comment from [simo](https://pagure.io/user/simo) at [2015-02-10 15:06:50](https://pagure.io/SSSD/sssd/issue/1527#230293)**

We spoke with Simo about NTLM auth and came up with a different idea. The named keyring might be nice to have but I'm inclined to defer the work.

changelog:  => 
design:  => 
design_review:  => 0
fedora_test_page:  => 
milestone: SSSD 1.13 beta => SSSD 1.13 backlog
priority: major => trivial
review:  => 0

---

**Comment from [simo](https://pagure.io/user/simo) at [2015-02-12 20:36:04](https://pagure.io/SSSD/sssd/issue/1527#230294)**

Mass-moving tickets not planned for any immediate release and re-setting priority.

milestone: SSSD 1.13 backlog => SSSD Deferred
priority: trivial => major

---

**Comment from [simo](https://pagure.io/user/simo) at [2016-11-17 11:31:09](https://pagure.io/SSSD/sssd/issue/1527#230295)**

This will be solved by KCM.

blockedby:  => 2887
sensitive:  => 0

---

**Comment from [simo](https://pagure.io/user/simo) at [2017-02-24 14:41:40](https://pagure.io/SSSD/sssd/issue/1527#230296)**

**Metadata Update from @simo**:
- Issue marked as depending on: #2887
- Issue set to the milestone: SSSD Patches welcome

---

**Comment from [simo](https://pagure.io/user/simo) at [2017-03-03 11:56:15](https://pagure.io/SSSD/sssd/issue/1527#429230)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset
- Custom field mark reset
- Custom field patch reset
- Custom field review reset
- Custom field sensitive reset
- Custom field testsupdated reset
- Issue close_status updated to: None

---

**Comment from [simo](https://pagure.io/user/simo) at [2020-03-24 14:20:55](https://pagure.io/SSSD/sssd/issue/1527#636945)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [simo](https://pagure.io/user/simo) at [2020-03-24 14:20:57](https://pagure.io/SSSD/sssd/issue/1527#636946)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
